src/qtism/runtime/tests/BasicSelection.php

```diff
@@ -74,12 +74,13 @@ public function select()
 
                     if ($withReplacement === false) {
                         unset($selectionBag[$baseIndex]);
-                        $selectionBag = array_values($selectionBag);
```

Comment:
Line 95 has the same issue, I believe.

WDYT of reversing the logic here, and populating the `$selectionBag` values in a foreach loop, while iterating over `$selectableRoutes`,  instead of setting all the values there, and then unsetting the unwanted ones?

Reply:
I believe Line 95 doesn't have this issue, because the index to access the reindexed array is calculated on each iteration, and not fixed by a `foreach`.
I would keep the overall implementation the same, for me it makes sense to pick out the `required` items and then randomly pick as much as needed from the leftovers